horovod/common/controller.cc

```diff
@@ -121,6 +128,9 @@ ResponseList Controller::ComputeResponseList(std::atomic_bool& shut_down,
     // a shutdown. This function removes any invalid cache entries, if they
     // exist.
     CoordinateCacheAndState(cache_coordinator);
+    if (state.joined && !cache_coordinator.invalid_bits().empty()) {
+      cache_coordinator.set_uncached_in_queue(true);
+    }
```

Comment:
This code here is setting the uncached_in_queue flag *after* the cache_coordinator has been synced in `CoordinateCacheAndState`, which is not good to do. However, I don't think this code is necessary. In the case where there is an invalid bit set, `uncached_in_queue` is already set during the processing of local requests in this else case here: https://github.com/horovod/horovod/blob/0d34859957b79ab07fe39bb4f755cefa7d84c986/horovod/common/controller.cc#L89
If there are any entries to be invalidated, this `uncached_in_queue` should already be true.

Reply:
Thanks. Definitely not needed, forgot to delete after some experiments. Removed.